include/deal.II/grid/tria.h

```diff
@@ -1468,6 +1474,18 @@ class Triangulation : public Subscriptor
     distorted_cells;
   };
 
+  /**
+   * If add_periodicity() is called, this variable stores the given
+   * periodic face pairs on level 0 for later access during the
+   * identification of ghost cells for the multigrid hierarchy and for
+   * setting up the periodic_face_map.
+   */
+  std::vector<GridTools::PeriodicFacePair<cell_iterator> > periodic_face_pairs_level_0;
+
+  /**
+   * If add_periodicity() is called, this variable stores the active periodic face pairs.
+   */
+  std::map<std::pair<cell_iterator, unsigned int>, std::pair<std::pair<cell_iterator, unsigned int>, std::bitset<3> > > periodic_face_map;
```

Comment:
I am fine with using std_cxx11::tuple<std::pair<...>,std::pair<...>,std::bitset<3> > if you like this better.